### PR TITLE
WIP(js): Convert everything to AMD

### PR DESCRIPTION
**Pull request by ewinslow**
BREAKING CHANGE:
elgg.provide and elgg.require have been removed.
Use `define` and `require` instead.
Core libraries are only available under AMD now.

### CODE REVIEW

File: mod/embed/views/default/js/elgg/embed.js.php [hunk 1]
```diff
@@ -0,0 +1,189 @@
+// <script>
+define(function(require) {
+	// inserts the embed content into the textarea
+	$(".embed-item").live('click', embed.insert);
```
Comment (mrclay) [inline mod/embed/views/default/js/elgg/embed.js.php]: Where are embed.insert and these others defined?

File: mod/embed/views/default/js/elgg/embed.js.php [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	 * @returns {String|Boolean}
+	 * @private
+	 */
+	embed._deprecated_custom_insert_js = function(hook, type, params, value) {
+		var textAreaId = params.textAreaId;
+		var content = params.content;
+		var event = params.event;
+	<?php
+		if (elgg_view_exists('embed/custom_insert_js')) {
+			elgg_deprecated_notice("The view embed/custom_insert_js has been replaced by the js hook 'embed', 'editor'.", 1.9);
```
Comment (mrclay) [inline mod/embed/views/default/js/elgg/embed.js.php]: Might as well ditch this, huh?

File: mod/embed/views/default/js/elgg/embed.js.php (hunk 1 above)
Comment (mrclay) [inline mod/embed/views/default/js/elgg/embed.js.php]: Problem is they aren't defined *yet*

File: js/lib/ajax.js [hunk 3]
```diff
@@ -1,261 +1,266 @@
...
-
-	//elgg.ajax('example/file.php', {data:{...}});
-	if (options.data) {
-		data_only = false;
-	} else {
-		for (member in options) {
-			//elgg.ajax('example/file.php', {callback:function(){...}});
-			if (elgg.isFunction(options[member])) {
-				data_only = false;
+define(function(require) {
```
Comment (mrclay) [inline js/lib/ajax.js]: This file doesn't need to be moved?

Comment (ewinslow) [inline js/lib/ajax.js]: Yea, it should, but you have an Elgg/ajax module already so I'm sorry of unsure where to put it.

**Comment by ewinslow** [comment]
Don't think this is going to happen even for 2.0. Too much breakage and pain...

It helps a little now that AMD is the only way to write inline JS, so there's that. Hopefully that will push folks to convert over...

Thank you to those kind souls converting their JS to AMD now, we may eventually get ourselves out of implicit-dependency/global-state madness.

**Pull request closed without merging**